auto_submit/lib/service/validation_service.dart

```diff
@@ -477,36 +481,25 @@ class ProcessMergeResult {
 typedef RetryHandler = Function();
 
 /// Internal wrapper for the logic of merging a pull request into github.
-Future<graphql.QueryResult> _processMergeInternal({
-  required graphql.GraphQLClient client,
-  required Config config,
-  required QueryResult queryResult,
-  required github.PullRequest messagePullRequest,
+Future<github.PullRequestMerge> _processMergeInternal(
+  GithubService githubService,
+  github.RepositorySlug slug,
+  int number,
+  github.MergeMethod mergeMethod, {
+  String? commitMessage,
+  String? requestSha,
 }) async {
-  final String id = queryResult.repository!.pullRequest!.id!;
-
-  final PullRequest pullRequest = queryResult.repository!.pullRequest!;
-  final Commit commit = pullRequest.commits!.nodes!.single.commit!;
-  final String? sha = commit.oid;
-  final int number = messagePullRequest.number!;
-
-  final graphql.QueryResult result = await client.mutate(
-    graphql.MutationOptions(
-      document: mergePullRequestMutation,
-      variables: <String, dynamic>{
-        'id': id,
-        'oid': sha,
-        'title': '${queryResult.repository!.pullRequest!.title} (#$number)',
-      },
-    ),
+  github.PullRequestMerge pullRequestMerge = await githubService.mergePullRequest(
```

Comment:
Nit: final

Do we expect to get some analysis complain due to `prefer_final_fields: true`?

Reply:
Yeah good catch. I see the analysis option for it. But for some reason it is not flagging this. I wonder if it is because it has the package alias on the object type?

Reply:
This does not appear to be working currently. Is this a dart team thing?

Reply:
Odd that did the trick. So we have several linter rules in the cocoon project itself. Should these be added to the top level rules?

Reply:
We want `prefer_final_fields` for all projects, but others may just fit auto_submit one. Can we move this one for now to unblock, and at the same time open a chat/bug with dart team to track?

Reply:
Done.